src/windows/wslaservice/inc/wslaservice.idl

```diff
@@ -116,9 +204,41 @@ struct _WSLA_SESSION_SETTINGS {
 ]
 interface IWSLASession : IUnknown
 {
+    // Image management.
+    HRESULT PullImage([in] LPCWSTR Image, [in, unique] struct WSLA_REGISTRY_AUTHENTICATION_INFORMATION* RegistryInformation, [in, unique] IProgressCallback* ProgressCallback);
+    HRESULT ListImages([out, size_is(, *Count)] struct WSLA_IMAGE_INFORMATION** Images, [out] ULONG* Count);
+    HRESULT DeleteImage([in] LPCWSTR Image);
+
+    // Container management.
+    HRESULT CreateContainer([in] struct WSLA_CONTAINER_OPTIONS* Options);
```

Comment:
Curious what does this do internally. In initial implementation:
1. Does this just map to nerdctl run? And the later StartContainer just does no-op if it's already running? Or
2. This one just does some sanity check, and the StartContainer does the actual nerdctl run

Reply:
This hasn't been decided yet, but it's most likely going to be one of those two in the first implementation yes

Reply:
`nerdctl run` is a shortcut for `nerdctl create` + `nerdctl start`